packages/assets-controllers/src/MultichainAssetsRatesController/MultichainAssetsRatesController.ts

```diff
@@ -42,11 +46,21 @@ import type {
  */
 const controllerName = 'MultichainAssetsRatesController';
 
+// This is temporary until its exported from snap
+type HistoricalPrice = {
+  intervals: HistoricalPriceIntervals;
+  // The UNIX timestamp of when the historical price was last updated.
+  updateTime: number;
+  // The UNIX timestamp of when the historical price will expire.
+  expirationTime?: number;
+};
+
 /**
  * State used by the MultichainAssetsRatesController to cache token conversion rates.
  */
 export type MultichainAssetsRatesControllerState = {
   conversionRates: Record<CaipAssetType, AssetConversion>;
+  historicalPrices: Record<CaipAssetType, Record<string, HistoricalPrice>>; // string being the current currency we fetched historical prices for
```

Comment:
A future improvement can be to clear the values that has an old expiration time

Reply:
Does this mean that the `historicalPrices` object will grow indefinitely?

Reply:
Great point. I think it will only infinitely grow when:
1. For each asset added (as it looks like we update existing assets)
2. For when the extension has not restarted (e.g. closing/opening browser) to clear this in-mem store.